clients/include/host_pinned_vector.hpp

```diff
@@ -0,0 +1,99 @@
+//
```

Comment:
Do we need a separate class for pinned memory? Or can we just pin memory on a `host_vector`, using `mlock()`?

Pinning memory seems an orthogonal concern to data layout. We do not need custom allocators or data structures to pin the memory. We can just add a function to our other classes to "pin the memory owned by this class".

Reply:
I think we want consistency with the hip API into which we are passing the memory.  Memory flags may be used by hip to determine what needs to be done, maybe it shouldn't be called pinned memory but hipHostAllocator.   I don't think we want to directly use low level memory bindings such as mlock which will be brittle if hip changes what they do behind the scenes, or they add windows support, or want to manage over allocations.  The use case is just for hipMemcpyAsync.   I can rename the class but now thinking I'll just merge what I have that works and can later see if other memory constraints arise in the next release and refactor when we get additional requirements.   For now this is only to be used in testing.